### PR TITLE
config cancan

### PR DESCRIPTION
**Pull request by nguyenngoctruong**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -4,7 +4,18 @@ class ApplicationController < ActionController::Base
...
   before_filter :configure_permitted_parameters, if: :devise_controller?
 
-   protected
+  rescue_from CanCan::AccessDenied do |exception|
+    redirect_to root_url, :alert => exception.message
+  end
+
+  def require_admin
+    unless current_user.admin?
+      flash[:danger] = "You are not admin"
```
Comment (nguyenduyta) [inline app/controllers/application_controller.rb]: I18n

File: app/models/user.rb [hunk 2]
```diff
@@ -22,4 +24,11 @@ def avatar_size
       errors.add :avatar, "should be less than 5MB"
     end
   end
+
+  def update_before_save
+    self.email = email.downcase
+    if role.nil?
+      self.role = "normal"
```
Comment (nguyenduyta) [inline app/models/user.rb]: self.normal!

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -4,6 +4,17 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_filter :configure_permitted_parameters, if: :devise_controller?
 
+  rescue_from CanCan::AccessDenied do |exception|
+    redirect_to root_url, :alert => exception.message
+  end
+
+  def require_admin
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: private method

File: app/views/users/_user.html.erb [hunk 4]
```diff
@@ -1,5 +0,0 @@
-<div>
```
Comment (dieunb) [inline app/views/users/_user.html.erb]: checkout this file

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -1,7 +0,0 @@
-<% provide :title, 'All users' %>
```
Comment (dieunb) [inline app/views/users/index.html.erb]: checkout this file

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -4,6 +4,18 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_filter :configure_permitted_parameters, if: :devise_controller?
 
+  rescue_from CanCan::AccessDenied do |exception|
+    redirect_to root_url, :alert => exception.message
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: coding convention

**Pull request merged**